Added backticks to spath keys and enabled the tests

Abigael-JT opened the pull request:
changes made:
- Added backticks to spath keys to escape dots in the path 
- Enabled eval spath test 
- changed expected column arrangement in aggregateAfterHdfsLoadTest

Related ticket
* #330

elliVM commented on src/test/java/com/teragrep/pth10/evalTest.java:
```diff
@@ -2609,7 +2609,7 @@ public void parseEvalSpathJSONCatalystTest() {
     // //person[age=30]/name/text()
     // FIXME broken due to spath udf changes
-    @Disabled
+    //@Disabled
```
Could this line be removed instead of commenting out?

elliVM reviewed (changes requested):
Applying of backticks could be done by an object that wraps the String with methods for formatted and original String values. Alternatively formatted Strings could be created at the start of the method with names like `stringWithBackticks` to clear the code a bit. This solution is also fine those changes are up to you.

### Check this please
Seems like some code format was changed. Please run  `mvn spotless:apply` to check for changes so it doesn't pop up on other branches. Otherwise looks good

Abigael-JT commented:
> Applying of backticks could be done by an object that wraps the String with methods for formatted and original String values. Alternatively formatted Strings could be created at the start of the method with names like `stringWithBackticks` to clear the code a bit. This solution is also fine those changes are up to you.
> ### Check this please
> 
> Seems like some code format was changed. Please run `mvn spotless:apply` to check for changes so it doesn't pop up on other branches. Otherwise looks good

Removed commented out code and added objects to escape and unescape backticks in spath keys, also added tests for those new objects. We also changed one of the spath test  to only filter out the duplicates and now there is only one assertion.

elliVM commented on src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java:
```diff
@@ -0,0 +1,46 @@
+package com.teragrep.pth10.steps.spath;import java.io.Serializable;
```
Add license. `mvn spotless:apply` should create missing licenses

elliVM commented on src/main/java/com/teragrep/pth10/steps/spath/SpathUnescapedKey.java:
```diff
@@ -0,0 +1,41 @@
+package com.teragrep.pth10.steps.spath;
```
Missing license

elliVM commented on src/test/java/com/teragrep/pth10/SpathEscapedKeyTest.java:
```diff
@@ -0,0 +1,54 @@
+package com.teragrep.pth10;
```
Missing license

elliVM commented on src/test/java/com/teragrep/pth10/SpathUnscapedKeyTest.java:
```diff
@@ -0,0 +1,53 @@
+package com.teragrep.pth10;
```
Missing license

elliVM reviewed (changes requested):
Changes look good to me just missing licenses and to apply spotless

elliVM reviewed (approved):
LGTM

eemhu commented:
Added missing SpathEscapedKey in evalStatement, rebased onto main and re-enabled disabled spath tests

kortemik commented on src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java:
```diff
@@ -0,0 +1,96 @@
+import java.io.Serializable;
+import java.util.Objects;
+
+public final class SpathEscapedKey implements Serializable {
```
SpathEscapedKey should have a method unescape that returns SpathKey

Abigael-JT replied:
This has been fixed

kortemik commented on src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java:
```diff
@@ -0,0 +1,94 @@
+        this.key = key;
+    }
+
+    public SpathKey unescaped() {
```
I was thinking this during the day ans thought that is this actually an escaped thing or is it rather a backtick quoted string and usual unquote functionality should.be used? @eemhu Any comments as you did previous quotation fixes

eemhu replied:
I suppose strictly speaking it is more of a backtick quotation but logically it is used to escape `.` (dots), so they can be used in the column names.

kortemik replied:
I think we should make the object re-usable for such purposes and use the generic quotation object that allows specifying the quotation character. this way it would be able to handle 
```
' " `
```
 and possibly other characters as well. current substring way does not handle properly escapes in a string. i.e. 
```
`foo\`bar`
```
 should be unquoted as 
```
foo`bar
```

eemhu replied:
Added backtick support for existing UnquotedText object and also added QuotedText object.

kortemik commented on src/main/java/com/teragrep/pth10/ast/UnquotedText.java:
```diff
@@ -53,7 +54,7 @@
  */
 public class UnquotedText implements Text {
```
i think this should take the quote character as an argument instead of using regex to match it?

eemhu replied:
Would it be okay if no quote character is provided it would remove all, and if it is provided it only removes the specific one? I think currently UnquotedText is used to remove all on purpose

kortemik reviewed (changes requested):
i would add a constructor parameter instead of using regex because regex is not very readable.

otherwise looks very clean and usable

kortemik commented:
good code: merging!
